Non-unified build fixes, early February 2024 edition

donny-dont opened the pull request:
#### 91c1a5ef6fbb412d867b06761133dfded7e286a5
<pre>
Non-unified build fixes, early February 2024 edition
<a href="https://bugs.webkit.org/show_bug.cgi?id=268592">https://bugs.webkit.org/show_bug.cgi?id=268592</a>

Unreviewed non-unified build fix.

* Source/WebCore/dom/EventTargetConcrete.cpp:
* Source/WebCore/fileapi/FileReader.cpp:
* Source/WebCore/html/VoidCallback.h:
* Source/WebCore/html/parser/HTMLResourcePreloader.cpp:
* Source/WebCore/html/track/VideoTrackList.cpp:
* Source/WebCore/layout/formattingContexts/inline/InlineContentCache.h:
* Source/WebCore/loader/LinkLoader.cpp:
* Source/WebCore/loader/MediaResourceLoader.cpp:
* Source/WebCore/loader/PingLoader.cpp:
* Source/WebCore/loader/TextTrackLoader.cpp:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
* Source/WebCore/rendering/svg/RenderSVGPath.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourcePattern.cpp:
* Source/WebCore/style/StylePendingResources.cpp:
* Source/WebCore/style/StyleScopeRuleSets.cpp:
* Source/WebCore/style/StyleSheetContentsCache.cpp:
* Source/WebCore/style/StyleSheetContentsCache.h:
* Source/WebCore/svg/SVGTitleElement.cpp:
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.cpp:
* Source/WebCore/xml/XMLHttpRequestUpload.cpp:
* Source/WebCore/xml/XSLStyleSheetLibxslt.cpp:
* Source/WebCore/xml/XSLTProcessorLibxslt.cpp:
* Source/WebKit/GPUProcess/GPUProcess.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
* Source/WebKit/GPUProcess/media/RemoteMediaResourceLoader.h:
* Source/WebKit/WebProcess/Cache/WebCacheStorageConnection.cpp:
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageBackendProxy.h:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaResourceProxy.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteVideoFrameProxy.cpp:

Canonical link: <a href="https://commits.webkit.org/274008@main">https://commits.webkit.org/274008@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/106a82a498ed774c027fc5c93709d4ce16dcfe72

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37540 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16425 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39804 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40066 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/33445 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38856 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19045 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13589 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31838 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38105 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13841 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/32924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12062 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/12027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/33610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41330 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33928 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/37922 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12598 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/10154 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36085 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/14046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8451 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/13003 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13359 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebCore/html/VoidCallback.h:
```diff
@@ -27,6 +27,7 @@
 
 #include "ActiveDOMCallback.h"
 #include "CallbackResult.h"
+#include "ContextDestructionObserverInlines.h"
```
If you include *Inlines.h in a header without Inlines.h suffix, it spoils the merit of splitting ContextDestructionObserverInlines.h and ContextDestructionObserver.h.

fujii replied:
But, this is not the only one including ContextDestructionObserverInlines.h from a header without Inlines.h suffix. Maybe, it's acceptable.

cdumez replied:
Probably OK in this case.

donny-dont committed:
Non-unified build fixes, early February 2024 edition
https://bugs.webkit.org/show_bug.cgi?id=268592

Unreviewed non-unified build fix.

* Source/WebCore/dom/EventTargetConcrete.cpp:
* Source/WebCore/fileapi/FileReader.cpp:
* Source/WebCore/html/VoidCallback.h:
* Source/WebCore/html/parser/HTMLResourcePreloader.cpp:
* Source/WebCore/html/track/VideoTrackList.cpp:
* Source/WebCore/layout/formattingContexts/inline/InlineContentCache.h:
* Source/WebCore/loader/LinkLoader.cpp:
* Source/WebCore/loader/MediaResourceLoader.cpp:
* Source/WebCore/loader/PingLoader.cpp:
* Source/WebCore/loader/TextTrackLoader.cpp:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
* Source/WebCore/rendering/svg/RenderSVGPath.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourcePattern.cpp:
* Source/WebCore/style/StylePendingResources.cpp:
* Source/WebCore/style/StyleScopeRuleSets.cpp:
* Source/WebCore/style/StyleSheetContentsCache.cpp:
* Source/WebCore/style/StyleSheetContentsCache.h:
* Source/WebCore/svg/SVGTitleElement.cpp:
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.cpp:
* Source/WebCore/xml/XMLHttpRequestUpload.cpp:
* Source/WebCore/xml/XSLStyleSheetLibxslt.cpp:
* Source/WebCore/xml/XSLTProcessorLibxslt.cpp:
* Source/WebKit/GPUProcess/GPUProcess.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
* Source/WebKit/GPUProcess/media/RemoteMediaResourceLoader.h:
* Source/WebKit/WebProcess/Cache/WebCacheStorageConnection.cpp:
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageBackendProxy.h:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaResourceProxy.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteVideoFrameProxy.cpp:

Canonical link: https://commits.webkit.org/274008@main
